Regenerate API docs (#796)
netobserv/network-observability-operator: docs/flowcollector-flows-netobserv-io-v1beta2.adoc

| List of additional features to enable. They are all disabled by default. Enabling additional features might have performance impacts. Possible values are: +

- `PacketDrop`: enable the packets drop flows logging feature. This feature requires mounting
the kernel debug filesystem, so the eBPF agent pods have to run as privileged.
If the `spec.agent.ebpf.privileged` parameter is not set, an error is reported. +

- `DNSTracking`: enable the DNS tracking feature. +

- `FlowRTT`: enable flow latency (sRTT) extraction in the eBPF agent from TCP traffic. +

- `NetworkEvents`: enable the network events monitoring feature, such as correlating flows and network policies. This feature requires mounting
the kernel debug filesystem, so the eBPF agent pods have to run as privileged.

> 

Reviewer: @jotak @msherif1234 so should i just manually remove this from the product docs when I copy it over?

Author: yeah actually I can also do it right here

Author: (it's part of 1.7 but won't work yet with current openshift)

> | List of additional features to enable. They are all disabled by default. Enabling additional features might have performance impacts. Possible values are: +

Reviewer: You're right, this was an incorrect adjustment on my part and peer review. I've reverted in my new PR: https://github.com/openshift/openshift-docs/pull/84158/files#diff-a8af995c8d95ce3ea42a525d345c161250163957c577c0daddbdcd1e0e48d2f6R205

| `certKey`
| `string`
| `certKey` defines the path to the certificate private key file name within the config map or secret. Omit when the key is not necessary.

| `name`
| `string`
| Name of the config map or secret containing certificates.

| `namespace`
| `string`
| Namespace of the config map or secret containing certificates. If omitted, the default is to use the same namespace as where Network Observability is deployed.
If the namespace is different, the config map or the secret is copied so that it can be mounted as required.

| `type`
| `string`
| Type for the certificate reference: `configmap` or `secret`.

|===
== .spec.exporters[].openTelemetry
Description::
+
--
OpenTelemetry configuration, such as the IP address and port to send enriched logs or metrics to.
--

Type::
  `object`

Required::
  - `targetHost`
  - `targetPort`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `fieldsMapping`
| `array`
| Custom fields mapping to an OpenTelemetry conformant format.
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal .

> As there is currently no accepted standard for L3/4 enriched network logs, you can freely override it with your own.

Reviewer: ```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can freely override it with your own.
```

Author: done

| `headers`
| `object (string)`
| Headers to add to messages (optional)

| `logs`
| `object`
| OpenTelemetry configuration for logs.

| `metrics`
| `object`
| OpenTelemetry configuration for metrics.

| `protocol`
| `string`
| Protocol of the OpenTelemetry connection. The available options are `http` and `grpc`.

| `targetHost`
| `string`
| Address of the OpenTelemetry receiver

| `targetPort`
| `integer`
| Port for the OpenTelemetry receiver

| `tls`
| `object`
| TLS client configuration.

|===
== .spec.exporters[].openTelemetry.fieldsMapping
Description::
+
--
Custom fields mapping to an OpenTelemetry conformant format.
By default, Network Observability format proposal is used: https://github.com/rhobs/observability-data-model/blob/main/network-observability.md#format-proposal .

> As there is currently no accepted standard for L3/4 enriched network logs, you can freely override it with your own.

Reviewer: ```suggestion
As there is currently no accepted standard for L3 or L4 enriched network logs, you can freely override it with your own.
```

Author: done

| `string`
| Type for the certificate reference: `configmap` or `secret`.

|===
== .spec.networkPolicy
Description::
+
--
`networkPolicy` defines ingress network policy settings for Network Observability components isolation.
--

Type::
  `object`

[cols="1,1,1",options="header"]
|===
| Property | Type | Description

| `additionalNamespaces`
| `array (string)`
| `additionalNamespaces` contains additional namespaces allowed to connect to the Network Observability namespace.
It gives some flexibility in the network policy configuration, however should you need a more specific
configuration, you can disable it and install your own instead.

| `enable`
| `boolean`
| Set `enable` to `true` to deploy network policies on the namespaces used by Network Observability (main and privileged). It is disabled by default.
These network policies better isolate the Network Observability components to prevent undesired connections to them.

> We recommend you either enable it, or create your own network policy for Network Observability.

Author: I don't really want this change, it's still a recommendation, not a requirement

> We recommend you either enable it, or create your own network policy for Network Observability.

Reviewer: Ok it is reverted in my new PR

| `object`
| scheduling controls how the pods are scheduled on nodes.

| `secondaryNetworks`
| `array`
| Define secondary networks to be checked for resources identification.
In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.

> If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labelled.

Reviewer: ```suggestion
If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labeled.
```

Author: done

| `object`
| scheduling controls how the pods are scheduled on nodes.

| `secondaryNetworks`
| `array`
| Define secondary networks to be checked for resources identification.

> In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.

Reviewer: ```suggestion
To guarantee a correct identification, indexed values must form an unique identifier across the cluster.
```

> If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labeled.

Reviewer: ```suggestion
If the same index is used by several resources, those resources might be incorrectly labeled.
```

== .spec.processor.advanced.secondaryNetworks
Description::
+
--
Define secondary networks to be checked for resources identification.

> In order to guarantee a correct identification, it is important that the indexed values form an unique identifier across the cluster.

Reviewer: ```suggestion
To guarantee a correct identification, the indexed values must form an unique identifier across the cluster.
```

> If there are collisions in the indexes (same index used by several resources), those resources might be wrongly labeled.

Reviewer: ```suggestion
If the same index is used by several resources, those resources might be wrongly labeled.
```
